constellations/gemini-user/src/programs/programs.js

```diff
@@ -0,0 +1,90 @@
+"use strict";
+require("./programs.sass");
+import React from "react";
+import API from "../utils/api.js";
+import { sortedSemesterInsert } from "../utils/semesterUtils.js";
+import { ProgramCard } from "./programCard.js";
+
+export class ProgramsPage extends React.Component {
+    state = {
+        semesters: [],
+        programClassesMap: {},
+    };
+
+    componentDidMount = () => {
+        // TODO: Change to below line when published_at is removed from all tables except classes
+        // API.get("api/classesbysemesters?published=true").then((res) => {
```

Comment:
I believe the published part of the endpoint is ready. The published_at stuff have been removed